internal/checker/checker.go

```diff
@@ -27110,7 +27110,7 @@ func (c *Checker) getContextuallyTypedParameterType(parameter *ast.Node) *Type {
 		return nil
 	}
 	iife := ast.GetImmediatelyInvokedFunctionExpression(fn)
-	if iife != nil && len(iife.Arguments()) != 0 {
```

Comment:
The original code says:

```
if (iife && iife.arguments) {
```

Should this be `iife != nil && iife.Arguments() != nil`, given `Arguments` is nil-preserving?

Reply:
Hm, no, I tested and that doesn't work. I'm not sure I get why two behave differently.

Reply:
I have not tested that but there is a chance that `iife.arguments` is just always truthy in Strada.

Reply:
I think that's honestly a bug in the parser. I was already looking at this via #706; let me recheck that.

Reply:
Oh, yeah.

```
export interface CallExpression extends LeftHandSideExpression, Declaration {
    readonly kind: SyntaxKind.CallExpression;
    readonly expression: LeftHandSideExpression;
    readonly questionDotToken?: QuestionDotToken;
    readonly typeArguments?: NodeArray<TypeNode>;
    readonly arguments: NodeArray<Expression>;
}
```

`arguments` is truthy. I'm double checking this in the tests, though.

Reply:
Yeah, this passes:

![image](https://github.com/user-attachments/assets/4c1bc086-1800-4694-94aa-80f8a65c4061)


Funny!